shm/ship.cpp

```diff
@@ -0,0 +1,109 @@
+#include "ship.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <memory>
+#include <string>
+
+Ship::Ship() : id_(-1) {}
+
+Ship::Ship(uint32_t capacity, uint32_t maxCrew, uint32_t speed, const std::string& name, uint32_t id, Time* time)
+    : capacity_(capacity), maxCrew_(maxCrew), crew_(0), speed_(speed), name_(name), id_(id), time_(time) {
+    time_->registerObserver(this);
+}
+
+Ship::~Ship() {
+    time_->unregisterObserver(this);
+}
+
+void Ship::setName(const std::string& name) {
+    name_ = name;
+}
+
+Ship& Ship::operator-=(uint32_t crew) {
+    if (crew < crew_) {
+        crew_ -= crew;
+        return *this;
+    }
+    std::cerr << "Too few sailors!";
+    return *this;
+}
+
+Ship& Ship::operator+=(uint32_t crew) {
+    if (crew_ + crew <= maxCrew_) {
+        crew_ += crew;
+        return *this;
+    }
+    std::cerr << "Too many sailors!";
+    return *this;
+}
+
+uint32_t Ship::getId() const {
+    return id_;
+}
+
+std::string Ship::getName() const {
+    return name_;
+}
+
+uint32_t Ship::getSpeed() const {
+    return speed_;
+}
+
+uint32_t Ship::getCrew() const {
+    return crew_;
+}
+
+uint32_t Ship::getMaxCrew() const {
+    return maxCrew_;
+}
+
+uint32_t Ship::getCapacity() const {
```

Comment:
getters should be implemented in hpp

Reply:
This is only cmake/tests PR

Reply:
ah okay sorry

Reply:
and we are defining all getters (even if they are short and easy) and constructors in cpp file, @nauka-programowania-MA was talking about it somewhere